tapscript/send.go

```diff
@@ -648,16 +648,20 @@ func SignVirtualTransaction(vPkt *tappsbt.VPacket, signer Signer,
 		prevAssets[input.PrevID] = input.Asset()
 	}
 
-	// Create a Taproot Asset virtual transaction representing the asset
-	// transfer.
-	virtualTx, _, err := VirtualTx(newAsset, prevAssets)
-	if err != nil {
-		return err
-	}
-
 	for idx := range inputs {
 		input := inputs[idx]
 
+		// Create a Taproot Asset virtual transaction representing the
+		// asset transfer. We provide this input's sighash flag and
+		// index in order to construct a virtual transaction that
+		// commits to the desired parts of the asset transfer.
+		virtualTx, _, err := VirtualTx(
```

Comment:
Why do we need to make a new virtual tx for each input? As we're concerned with signing, isn't just `VirtualTxWithInput` enough?